labs/b4.md

```diff
@@ -0,0 +1,165 @@
+
+Debian is a distribution developed by the Debian Project beginning in 1993. This distribution strives to provide a completely free (libre) software experience. Many other distributions are based on Debian as Debian releases are usually very stable. One such distribution is Ubuntu, which is very popular and has spawned its many of its own derivatives. What unifies all of Debian-based distributions is the usage of the DEB package format and the dpkg package manager.
+
+## But what about these “package managers”?
+
+If you come from the Windows or even MacOS ecosystem, you might be scratching your head thinking what a package managers is. On those operating systems, the most conventional and used method of installing programs is to run an installer that copies the files from an archive to the correct place (and makes changes to a registry of some sorts).
+
+Enter the package manager, a centralized way to install, update, and remove software from repositories/sources. Most package managers work in the same way: the package manager gets or updates a list of packages and then fetches the packages that is referenced in the list if is requested to be installed. A package manager is like a librarian. When a patron wants to read a book, the librarian consults a catalog (which is the package database) and then fetches the book from the shelf and gives it to the patron (installing the package). Sometimes, the librarian has to update the catalog because new editions of the books were added (updating the package database).
+
+## Debian: An Example
+
+In this class, we will be focused on using Debian. As noted before, Debian uses apt/dpkg as its package manager.
+
+To package manager is invoked by using apt. Before you do anything, it is good package to update the package list so your package manager can find and fetch the most up-to-date packages. To do that, run:
+
+`apt update`
+
+To find the package to install:
+
+`apt cache search [package|description]`
+
+To install a package, run:
+
+`apt install package`
+
+To remove a package, run:
+
+`apt remove package`
+
+Easy? Sometimes, you want to update the packages that you have installed, run:
+
+`apt dist-upgrade`
+
+There are also other commands, such as removing unneeded dependencies and purging packages but that is what the man pages are for.
+
+### Example
+
+We are going to install GCC for the next step of the lab.
+
+**This, and all other parts of the lab, should be done on your DigitalOcean VMs, as you cannot install packages on OCF machines.** Please note that you can only connect to your VM from inside the Berkeley network, so you will have to either be on campus wifi, or you
+will have to SSH through `ssh.ocf.berkeley.edu` (or a similar on-campus
+host) to access your VM. You should be able to connect to it at `<OCF username>.decal.xcf.sh` by running `ssh <username>@<username>.decal.xcf.sh`.
+
+Simply run:
+
+`sudo apt install gcc`
+
+Now check if GCC is installed by typing the followng:
```

Comment:
Maybe explain that this will check the version of GCC? Probably doesn't matter though, that'll be pretty clear when they run the command I suppose.